gst/gst_device_provider_factory.go

```diff
@@ -0,0 +1,23 @@
+package gst
+
+// #include "gst.go.h"
+import "C"
+
+import (
+	"unsafe"
+)
+
+// DeviceProviderFactory is a Go representation of a GstDeviceProviderFactory.
+type DeviceProviderFactory struct {
+	ptr *C.GstDeviceProviderFactory
+}
+
+func FindDeviceProviderByName(factoryName string) *DeviceProvider {
+	cFactoryName := C.CString(factoryName)
+	defer C.free(unsafe.Pointer(cFactoryName))
+	provider := C.gst_device_provider_factory_get_by_name((*C.gchar)(unsafe.Pointer(cFactoryName)))
```

Comment:
Doesn't this need a `glib.TransferFull` to avoid memory leaks?

according to @tinyzimmer you need to look for the `transfer` attribute on the [return types from the docs](https://gstreamer.freedesktop.org/documentation/gstreamer/gstdeviceproviderfactory.html?gi-language=c#gst_device_provider_factory_get_by_name) and use the correct glib transfer functions.

related: #2

Reply:
It wouldn't surprise me if that was the case! I'll take a look :) before this I hadn't done much cgo so these pull requests definitely need a little work :)

Reply:
ok I think I fixed this up the right way :)